Implement named arguments for NEW DTO syntax (#11116)
doctrine/doctrine2: docs/en/reference/dql-doctrine-query-language.rst

Note that you can only pass scalar expressions to the constructor.

> The ``NEW`` operator also supports named arguments, similarly to php 8.0 :

Reviewer: ```suggestion
The ``NEW`` operator also supports named arguments:
```

.. code-block:: php

    <?php
    $query = $em->createQuery('SELECT NEW CustomerDTO(email: e.email, name: c.name, address: a.city) FROM Customer c JOIN c.email e JOIN c.address a');
    $users = $query->getResult(); // array of CustomerDTO

> Note that you cannot pass unnamed arguments after named ones, just like in php.

Reviewer: ```suggestion
Note that you must not pass ordered arguments after named ones.
```
